### PR TITLE
bug fix in addLeadingZeros (that occured with very large numbers)

### PR DESCRIPTION
**Pull request by sachseka**
I'd be glad if one of you could check this

### CODE REVIEW

File: R/addLeadingZeros.R [hunk 1]
```diff
@@ -14,7 +15,7 @@ addLeadingZerosToCharInt <- function(dat) {
   lenToCorrect <- unlist(lapply(dat, FUN=function(ii) { max(nchar(c(ii,"a")),na.rm=TRUE)}))
   act <- sapply(seq(along=dat), function(jj) all(elToCorrect[jj][[1]]) & !all(isna[jj][[1]]) & lenToCorrect[jj]>1)
    if(any(act)) {
-     dat[,act] <- data.frame(lapply(which(act), function(x) ifelse(is.na(dat[,x]), NA, formatC(as.numeric(dat[,x]), width = lenToCorrect[x], format = "d", flag = "0"))))
+     dat[,act] <- data.frame(lapply(which(act), function(x) ifelse(is.na(dat[,x]), NA, stringi::stri_pad(dat[, x], width = lenToCorrect[x], side = "left", pad = "0"))))
```
Comment (beckerbenj) [inline R/addLeadingZeros.R:18]: LGTM

Nit: Could this line be broken up in a conceptual helpful manner or at least line breaks added to increase readability?

File: tests/testthat/test_addLeadingZeros.R [hunk 2]
```diff
@@ -1,6 +1,7 @@
 
 ### set up test data frames
-dat <- data.frame(v1 = c("0","300","e",NA),
+dat <- data.frame(ID=c("112324234244","112324234245","131224234246","112324234247"),
```
Comment (beckerbenj) [inline tests/testthat/test_addLeadingZeros.R:3]: LGTM

However, line 10 seems redundant to line 20 to me? And the test title in line 19 is really confusing to me?

**Review by beckerbenj — CHANGES_REQUESTED** [review]
The fix looks good to me! See two specific minor comments inline and to general minor comments below. 

- Could a news entry be added for this change?
- The documentation is a bit misleading to an outsider (at least as I understand it). It speaks of adding leading zeros, but the way the description is written I would expect leading zeros to be added to all numbers (including the ID variable in the test). But I guess what the function does is add leading zeros so all numbers have the same number of digits? Maybe this is described under `\value`, but this seems rather hidden to me.

File: R/addLeadingZeros.R (hunk 1 above)
Comment (sachseka) [reply R/addLeadingZeros.R:18]: ok

File: tests/testthat/test_addLeadingZeros.R (hunk 2 above)
Comment (sachseka) [reply tests/testthat/test_addLeadingZeros.R:3]: Thank you for pointing this out! This test was ill-labeled, sorry. I will improve it also by adding additional tests.

**Comment by sachseka** [comment]
> The fix looks good to me! See two specific minor comments inline and to general minor comments below.
> 
>     * Could a news entry be added for this change?
sure
> 
>     * The documentation is a bit misleading to an outsider (at least as I understand it). It speaks of adding leading zeros, but the way the description is written I would expect leading zeros to be added to all numbers (including the ID variable in the test). But I guess what the function does is add leading zeros so all numbers have the same number of digits? Maybe this is described under `\value`, but this seems rather hidden to me.

ok i will try to improve the description

**Comment by sachseka** [comment]
I implemented the requested changes. Could you please re-check?

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,4 @@
+* minor bugfixes when integers are very large in `addLeadingZerosToCharInt()` (Previously an error message was displayed.)
```
Comment (beckerbenj) [inline NEWS.md:1]: LGTM

File: R/addLeadingZeros.R [hunk 4]
```diff
@@ -14,7 +15,12 @@ addLeadingZerosToCharInt <- function(dat) {
   lenToCorrect <- unlist(lapply(dat, FUN=function(ii) { max(nchar(c(ii,"a")),na.rm=TRUE)}))
   act <- sapply(seq(along=dat), function(jj) all(elToCorrect[jj][[1]]) & !all(isna[jj][[1]]) & lenToCorrect[jj]>1)
    if(any(act)) {
-     dat[,act] <- data.frame(lapply(which(act), function(x) ifelse(is.na(dat[,x]), NA, formatC(as.numeric(dat[,x]), width = lenToCorrect[x], format = "d", flag = "0"))))
+     dat[,act] <- data.frame(lapply(which(act), function(x) {
```
Comment (beckerbenj) [inline R/addLeadingZeros.R:18]: LGTM

File: man/addLeadingZeros.Rd [hunk 5]
```diff
@@ -11,7 +11,7 @@ a data.frame consisting of character columns only
 }
 }
 \value{
-a data.frame of only character columns and the same dimensions as the input data.frame where all columns with integers are all of the same arity now due to added leading zeros.
+a data.frame of only character columns and the same dimensions as the input data.frame. In any column containing strings that can be converted to integers, these strings will be padded with leading zeros so that all values in the column have the same number of digits.
```
Comment (beckerbenj) [inline man/addLeadingZeros.Rd:14]: LGTM

File: tests/testthat/test_addLeadingZeros.R (hunk 2 above)
Comment (beckerbenj) [reply tests/testthat/test_addLeadingZeros.R:3]: LGTM

File: tests/testthat/test_addLeadingZeros.R [hunk 6]
```diff
@@ -15,9 +15,9 @@ test_that("Errors", {
                "All columns in dat need to be of class character.")
 })
 
-test_that("Factor vector drop levels", {
+test_that("Normal functioning", {
```
Comment (beckerbenj) [inline tests/testthat/test_addLeadingZeros.R:18]: LGTM

**Review by beckerbenj — APPROVED** [review]
All changes look good to me!

One last request: Could you up the version number in the description to 0.7.7.9000?

**Comment by sachseka** [comment]
> All changes look good to me!
> 
> One last request: Could you up the version number in the description to 0.7.7.9000?

Sure, I'll update the version number to 0.7.7.9000. Thanks for reviewing the changes!

**Pull request merged**